examples/aco_tsp/aco_tsp/model.py

```diff
@@ -0,0 +1,249 @@
+            return current_city
+
+        # p_ij(t) = 1/Z*[(tau_ij)**alpha * (1/distance)**beta]
+        results = []
+        for city in candidates:
+            val = (
+                (g[current_city][city]["pheromone"]) ** self.alpha
+                * (g[current_city][city]["visibility"]) ** self.beta
+            )
+            results.append(val)
+
+        results = np.array(results)
+        Z = results.sum()
+        results /= Z
+
+        new_city = self.model.random.choices(candidates, weights=results)[0]
+
+        return new_city
+
+    def step(self):
+        """
+        Modify this method to change what an individual agent will do during each step.
+        Can include logic based on neighbors states.
+        """
+        g = self.model.grid.G
+        for idx in range(self.model.num_cities - 1):
+            # Pick a random city that isn't in the list of cities visited
+            current_city = self.pos
+            new_city = self.decide_next_city()
+            self._cities_visited.append(new_city)
+            self.model.grid.move_agent(self, new_city)
+            self._traveled_distance += g[current_city][new_city]["distance"]
+
+        self.tsp_solution = self._cities_visited.copy()
+        self.tsp_distance = self._traveled_distance
+        self._cities_visited = []
+        self._traveled_distance = 0
+
+
+class AcoTspModel(mesa.Model):
+    """
+    The model class holds the model-level attributes, manages the agents, and generally handles
+    the global level of our model.
+
+    There is only one model-level parameter: how many agents the model contains. When a new model
+    is started, we want it to populate itself with the given number of agents.
+
+    The scheduler is a special model component which controls the order in which agents are activated.
+    """
+
+    def __init__(
+        self,
+        num_agents: int = 20,
+        tsp_graph: TSPGraph = TSPGraph.from_random(20),
+        max_steps: int = int(1e6),
+        ant_alpha: float = 1.0,
+        ant_beta: float = 5.0,
+    ):
+        super().__init__()
+        self.num_agents = num_agents
+        self.tsp_graph = tsp_graph
+        self.num_cities = tsp_graph.num_cities
+        self.all_cities = set(range(self.num_cities))
+        self.max_steps = max_steps
+        self.schedule = mesa.time.RandomActivation(self)
+        self.grid = mesa.space.NetworkGrid(tsp_graph.g)
+
+        for i in range(self.num_agents):
+            agent = AntTSP(unique_id=i, model=self, alpha=ant_alpha, beta=ant_beta)
+            self.schedule.add(agent)
+
+            city = tsp_graph.cities[self.random.randrange(self.num_cities)]
+            self.grid.place_agent(agent, city)
+            agent._cities_visited.append(city)
+
+        self.num_steps = 0
+        self.best_path = None
+        self.best_distance = float("inf")
+        self.best_distance_iter = float("inf")
+
+        self.datacollector = mesa.datacollection.DataCollector(
```

Comment:
Maybe describe in the Readme briefly what data is collected and how that is used or can be used to gain insights into the system.